pkg/util/provider/machinecontroller/machine_util.go

```diff
@@ -1356,3 +1519,16 @@ func getProviderID(machine *v1alpha1.Machine) string {
 func getNodeName(machine *v1alpha1.Machine) string {
 	return machine.Status.Node
 }
+
+func getMachineOwnerNames(machine *v1alpha1.Machine) (string, string) {
+	machineSetName := machine.OwnerReferences[0].Name
+	var machineDeployName string
+	msLen := len(machineSetName)
+	for i := msLen - 1; i >= 0; i-- {
+		if machineSetName[i] == '-' {
+			machineDeployName = machineSetName[:i]
+			break
+		}
+	}
```

Comment:
I would recommend invoking a `Get `call on the `machineSet` and referring the owner reference of this `machineSet `to fetch the name of `MachineDeployment` instead of string handling with `-` as delimiter.

Reply:
Hey Samarth :D 
Yeah I was thinking the same, but since this is a part of MC , so I would have to create a seperate clientset for machineDeployments , and since MC shouldn't be aware of machineDeployment so I wrote this way.

Reply:
Oh! That's quite tricky. In that case, I have another "more Golang" way of putting this up as a code as below.

```suggestion
	machineSetNameParts := strings.Split(machineSetName, "-")
	machineDeployName := strings.Join(machineSetNameParts[:len(machineSetNameParts)-1], "-")
```

Verify the index logic once. I have directly put up the code here.

Reply:
Thanks Samarth, it looks quite clean now. I have verified its working correctly!